tools/LoadPackage.groovy

```diff
@@ -234,23 +251,26 @@ class LoadPackage {
 
             // POST QUERY
             def query_push_obj_query = this.CATALOG_URL + "/buckets/" + bucket_name + "/resources?name=" + object_name + "&kind=" + kind + "&commitMessage=" + commitMessageEncoded + "&objectContentType=" + contentType
-            def post = new org.apache.http.client.methods.HttpPost(query_push_obj_query)
-            post.addHeader("Accept", "application/json")
-            post.addHeader("Content-Type", org.apache.http.entity.ContentType.MULTIPART_FORM_DATA.getMimeType() + ";boundary=" + boundary)
-            post.addHeader("sessionId", this.sessionId)
+            createAndExecuteQueryWithFileAttachment(query_push_obj_query, boundary, object_file)
+            writeToOutput(" " + object_file.getName() + " created!")
+        } else {
+            // Retrieve object metadata
+            def commitMessageEncoded = java.net.URLEncoder.encode(metadata_map.get("commitMessage") + " (" + SCHEDULER_VERSION + ")", "UTF-8")
 
-            def builder = org.apache.http.entity.mime.MultipartEntityBuilder.create()
-            builder.setBoundary(boundary);
-            builder.setMode(org.apache.http.entity.mime.HttpMultipartMode.BROWSER_COMPATIBLE)
-            builder.addPart("file", new org.apache.http.entity.mime.content.FileBody(object_file))
-            post.setEntity(builder.build())
+            // For POST queries
+            this.class.getClass().getResource(new File(this.SCHEDULER_HOME, "dist/lib/httpclient-4.5.2.jar").absolutePath);
+            def boundary = "---------------" + UUID.randomUUID().toString();
```

Comment:
why do we generate boundary like this?

Reply:
I have no idea honestly, I did not want to touch this bit